xrayvision/visibility.py

```diff
@@ -5,10 +5,408 @@
+    @property
+    @abc.abstractmethod
+    def observer_coordinate(self) -> Union[SkyCoord, None]:
+        """
+        Location of the observer.
+        """
+
+    @property
+    @abc.abstractmethod
+    def energy_range(self) -> Union[Iterable[apu.Quantity], None]:
+        """
+        Energy range over which the visibilities are computed.
+        """
+
+    @property
+    @abc.abstractmethod
+    def time_range(self) -> Union[Iterable[Time], None]:
+        """
+        Time range over which the visibilities are computed.
+        """
+
+    @property
+    @abc.abstractmethod
+    def vis_labels(self) -> Union[Iterable[str], None]:
+        """
+        Labels of each visibility.
+        """
+
+    @property
+    @abc.abstractmethod
+    def instrument(self) -> Union[str, None]:
+        """
+        The name of the instrument or observer that measured the visibilities.
+        """
+
+
+class VisibilitiesABC(abc.ABC):
+    @property
+    @abc.abstractmethod
+    def visibilities(self) -> Iterable[apu.Quantity]:
+        """
+        Complex numbers representing the visibilities.
+        """
+
+    @property
+    @abc.abstractmethod
+    def u(self) -> Iterable[apu.Quantity]:
+        """
+        u-coordinate on the complex plane of the visibilities.
+        """
+
+    @property
+    @abc.abstractmethod
+    def v(self) -> Iterable[apu.Quantity]:
+        """
+        v-coordinate on the complex plane of the visibilities.
+        """
+
+    @property
+    @abc.abstractmethod
+    def phase_center(self) -> apu.Quantity[apu.deg]:
+        """
+        The position of the phase center of the visibilities.
+        """
+
+    @property
+    @abc.abstractmethod
+    def meta(self) -> VisMetaABC:
+        """
+        Metadata.
+        """
+
+    @property
+    @abc.abstractmethod
+    def uncertainty(self) -> Union[Iterable[apu.Quantity], None]:
+        """
+        Uncertainties on visibilities values.
+        """
+
+    @property
+    @abc.abstractmethod
+    def amplitude(self) -> Union[Iterable[apu.Quantity], None]:
+        """
+        Amplitudes of the visibilities.
+        """
+
+    @property
+    @abc.abstractmethod
+    def amplitude_uncertainty(self) -> Union[Iterable[apu.Quantity], None]:
+        """
+        Amplitude uncertainty of the visibilities.
+        """
+
+    @property
+    @abc.abstractmethod
+    def phase(self) -> Union[Iterable[apu.Quantity[apu.deg]], None]:
+        """
+        Phases of the visibilities.
+        """
+
+    @property
+    @abc.abstractmethod
+    def phase_uncertainty(self) -> Union[Iterable[apu.Quantity[apu.deg]], None]:
+        """
+        Phase uncertainty of the visibilities.
+        """
+
+
+class Visibilities(VisibilitiesABC):
+    @apu.quantity_input()
+    def __init__(
+        self,
+        visibilities: apu.Quantity,
+        u: apu.Quantity[1 / apu.deg],
+        v: apu.Quantity[1 / apu.deg],
+        phase_center: Union[apu.Quantity[apu.deg], None] = None,
```

Comment:
I'm not sure but I think when you have `quantity_input()` decorator and somethine like `apu.Quantity[apu.deg]` it will try convert the input to `deg`. Not sure how it work with in a `Union`

Reply:
I tested it out locally and it didn't actually convert the unit.